packages/connect-iframe/src/systemInfo.ts

```diff
@@ -44,72 +42,27 @@ const getSuggestedPlatform = (): Platform | undefined => {
     }
 };
 
-type BrowserState = {
-    name: string;
-    osname: string;
-    supported: boolean;
-    outdated: boolean;
-    mobile: boolean;
-};
-
-const DEFAULT_STATE: BrowserState = {
-    name: 'unknown',
-    osname: 'unknown',
-    supported: false,
-    outdated: false,
-    mobile: false,
-};
-
-type SupportedBrowser = {
-    version: number;
-    download: string;
-    update: string;
-};
-
-const getBrowserState = (supportedBrowsers: { [key: string]: SupportedBrowser }): BrowserState => {
-    if (typeof window === 'undefined' || !navigator || !navigator.userAgent) return DEFAULT_STATE;
-    const browser = {
-        version: getBrowserVersion(),
-        name: getBrowserName(),
-    };
-    const os = {
-        version: getOsName(),
-        name: getOsVersion(),
-    };
-
+export const getSystemInfo = (supportedBrowsers: {
+    [key: string]: { version: number };
+}): SystemInfo => {
+    const browserName = getBrowserName();
+    const browserVersion = getBrowserVersion();
+    const supportedBrowser = browserName ? supportedBrowsers[browserName.toLowerCase()] : undefined;
```

Comment:
these are the possible browser names:

```
2345Explorer, 360 Browser, Amaya, Android Browser, Arora, Avant, Avast, AVG,
BIDUBrowser, Baidu, Basilisk, Blazer, Bolt, Brave, Bowser, Camino, Chimera,
Chrome Headless, Chrome WebView, Chrome, Chromium, Cobalt, Comodo Dragon, Dillo,
Dolphin, Doris, DuckDuckGo, Edge, Electron, Epiphany, Facebook, Falkon, Fennec, 
Firebird, Firefox [Focus/Reality], Flock, Flow, GSA, GoBrowser, Huawei Browser, 
ICE Browser, IE, IEMobile, IceApe, IceCat, IceDragon, Iceweasel, Instagram, 
Iridium, Iron, Jasmine, Kakao[Story/Talk], K-Meleon, Kindle, Klar, Konqueror, 
LBBROWSER, Line, LinkedIn, Links, Lunascape, Lynx, MIUI Browser, Maemo Browser, 
Maemo, Maxthon, MetaSr Midori, Minimo, Mobile Safari, Mosaic, Mozilla, NetFront, 
NetSurf, Netfront, Netscape, NokiaBrowser, Obigo, Oculus Browser, OmniWeb, 
Opera Coast, Opera [Mini/Mobi/Tablet], PaleMoon, PhantomJS, Phoenix, Polaris, 
Puffin, QQ, QQBrowser, QQBrowserLite, Quark, QupZilla, RockMelt, Safari, 
Sailfish Browser, Samsung Browser, SeaMonkey, Silk, Skyfire, Sleipnir, Slim, 
SlimBrowser, Swiftfox, Tesla, Tizen Browser, UCBrowser, UP.Browser, Viera, 
Vivaldi, Waterfox, WeChat, Weibo, Yandex, baidu, iCab, w3m, Whale Browser...
```
for example it has `Firefox [Focus/Reality]`not sure how it works in reality.